src/nvim/generators/gen_api_dispatch.lua

```diff
@@ -260,6 +260,13 @@ for i = 1, #functions do
       args[#args + 1] = converted
     end
 
+    if fn.check_textlock then
+      output:write('\n  if (textlock != 0) {')
+      output:write('\n    api_set_error(error, kErrorTypeException, "%s", e_secure);')
+      output:write('\n    goto cleanup;')
+      output:write('\n  }\n')
+    end
```

Comment:
📝 `text_locked()` checks `cmdwin_type`.
This should check only pure textlock.
Because the following should be able to use in cmdwin if not textlocked.

- `nvim_paste`
- `nvim_put`
- `nvim_set_current_line`
- `nvim_del_current_line`
- `nvim_buf_set_lines`
- `nvim_win_close`

So, the following may need checking cmdwin. (in separated pr)

- `nvim_set_current_buf`
- `nvim_set_current_win`
- `nvim_open_win`
- `nvim_set_current_tabpage`
- `nvim_win_set_buf`
- `nvim_buf_delete`